chain/queue.go

```diff
@@ -0,0 +1,88 @@
+package chain
+
+import (
+	"container/list"
+)
+
+// ConcurrentQueue is a concurrent-safe FIFO queue with unbounded capacity.
+// Clients interact with the queue by pushing items into the in channel and
+// popping items from the out channel. There is a goroutine that manages moving
+// items from the in channel to the out channel in the correct order that must
+// be started by calling Start().
+type ConcurrentQueue struct {
```

Comment:
Maybe not in the scope for this PR, but would be nice to extract this into its own package/repo, such that it could be shared between `lnd` and `btcwallet`,